lesson_14/exceptions/exceptions_app/src/main/java/com/codedifferently/lesson14/ecommerce/EcommerceSystem.java

```diff
@@ -17,19 +17,31 @@ public void addProduct(String productId, String name) {
     products.put(productId, new Product(productId, name));
   }
 
-  public String placeOrder(String productId, int quantity) {
+  public String placeOrder(String productId, int quantity) throws ProductNotFoundException {
     Product product = products.get(productId);
     String orderId = UUID.randomUUID().toString();
     orders.put(orderId, new Order(orderId, product, quantity));
+    // Asks if product is null and to throw the exception message 'Product with ID {orderId} not
+    // found' if it is.
+    if (product == null) {
```

Comment:
Odd to put it here since an empty product would probably lead the `Order` constructor to blow up. Check your inputs before you do any work in the function.